### ISSUE TITLE
Cannot move vertex to a new class or cluster

### ISSUE THREAD

**Issue by shooit** [orientechnologies/orientdb#7171]
## Expected behavior  
- `vertex.moveToClass()` should move the vertex to a new class
- `vertex.moveToCluster()` should move the vertex to a new cluster

## Actual behavior  
- The vertex is not moved to a new cluster
- The vertex is not moved to a new cluster
## Steps to reproduce  
```java
@Test
public void testMoveToClass(){
    String location = "memory:test_move_class";
    OrientGraphNoTx graphNoTx = new OrientGraphNoTx(location);
    OrientVertexType v2 = graphNoTx.createVertexType("V2");
    graphNoTx.shutdown();
    
    OrientGraph graph = new OrientGraph(location);
    OrientVertex v = graph.addVertex(null);
    assertEquals(graph.getVertexBaseType(), v.getType());
    v.moveToClass("V2");
    graph.commit();
    assertEquals(v2, v.getType()); //fails
}

@Test
public void testMoveCluster(){
    String location = "memory:test_move_cluster";
    OrientGraph graph = new OrientGraph(location);
    OrientVertex v = graph.addVertex("cluster:9");
    graph.commit();
    ORecordId original = (ORecordId) v.getId();
    v.moveToCluster("10");
    graph.commit();
    assertNotEquals(original, v.getId()); //fails
}
```

**Comment by shooit** [comment]
Actually this looks more like a case of mixed paradigms and incomplete documentation than a bug...

Doc-string for `vertex.moveTo` is "Moves current vertex to another class/cluster. All edges are updated automatically."

Looked into the implementation of `moveTo` and what is actually happening is closer to a functional paradigm as the original vertex is removed from the graph and a copy is made and placed into the new class or cluster. 

I was not expecting this behavior given that the rest of the methods of OrientVertex are modifying it in place as expected in the OO paradigm.

Is it possible to have this method update the vertex in place?

**Comment by luigidellaquila** [comment]
Hi @shooit 

The physical record cannot be updated, because its identity (RID) depends on which class/cluster it belongs to.
The OrientVertex is actually a wrapper of the record, so the problem could be easily fixed adding a 

```java 
this.rawElement = doc;
```

at the end of the moveTo(String, String) method. I'll talk to the team to see if it makes sense to change this behavior

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Hi @shooit 

I discussed it with the team, it definitely makes sense to update the internal references to point to the new vertex. 
I'm doing this change now, you will find it in next 2.2.18 (sorry, too late to have it in 2.2.17, as we are already in the middle of the release process)

Thanks

Luigi

**Comment by shooit** [comment]
@luigidellaquila Great! Thank you. I think this is great for consistency's sake.

**Comment by luigidellaquila** [comment]
Hi @shooit 

I just pushed a fix on branch 2.2.x, it will be released with 2.2.18

Thanks

Luigi
